src/main/scala/TypeClass.scala

```diff
@@ -0,0 +1,49 @@
+trait CanAdd[A] {
+  def plus(a: A, b: A): A
+  def apply(a: A, b: A): A = plus(a, b)
+}
+object CanAdd {
+  implicit def fromCanAddZ[A](implicit canAddZ: CanAddZ[A]): CanAdd[A] = canAddZ
```

Comment:
Comment this function, since its confusing? s/t like "If we have a Monoid[A], we also have a Semigroup[A]"

Reply:
done